utils/src/main/scala/za/co/absa/enceladus/utils/explode/ExplodeTools.scala

```diff
@@ -178,10 +179,13 @@ object ExplodeTools {
       case Some(errorCol) =>
         // Implode taking into account the error column
         // Errors should be collected, flattened and made distinct
+        // decDf.schema: "errCol: array (nullable = true)", while the result would contain "errCol: array (nullable = false)"
+        // bc. collect_list. See issue #1818
         decDf.orderBy(orderByRecordCol, orderByInsideArray)
           .groupBy(groupByColumns: _*)
           .agg(collect_list(deconstructedField).as(tmpColName),
             array_distinct(flatten(collect_list(col(errorCol)))).as(errorCol))
+          .withNullableColumnState(errorCol, true) // explicitly nullable: expected schema outcome, see issue #1818
```

Comment:
Shouldn't this be based on nullability configuration?

Reply:
On this level, it is deliberate to maintain consistency. (Spark 2 it was always nullable here, for Spark 3 we make it the same). Because setting it here by the configuration value (I have tried) has the effect of only for certain types of CRs being used (dubbed as _Option 2_ above)

Then, on the level of Standardization and Conformance as a whole, the nullability configuration is applied on that level, specifically:
```
DynamicInterpreter.interpret:74
StandardizationInterpreter.standardize:73
```

So we have internals that behave uniformly and externals that behave according to the configuration. By applying the config here, in ExplodeToools:184, too,  we would have forgo internal uniformity.

Reply:
Thanks for the explanation. 👍